interactions/client/bot.py

```diff
@@ -1539,6 +1541,162 @@ async def _logout(self) -> None:
         await self._websocket.close()
         await self._http._req.close()
 
+    async def wait_for(
+        self,
+        name: str,
+        check: Optional[Callable[..., Union[bool, Awaitable[bool]]]] = None,
+        timeout: Optional[float] = None,
+    ) -> Any:
+        """
+        Waits for an event once, and returns the result.
+
+        Unlike event decorators, this is not persistent, and can be used to only proceed in a command once an event happens.
+
+        :param str name: The event to wait for
+        :param Callable check: A function or coroutine to call, which should return a truthy value if the data should be returned
+        :param float timeout: How long to wait for the event before raising an error
+        :return: The value of the dispatched event
+        :rtype: Any
+        """
+        while True:
+            fut = self._websocket._dispatch.add(name=name)
+            try:
+                # asyncio's wait_for
+                res: tuple = await wait_for(fut, timeout=timeout)
+            except TimeoutError:
+                with contextlib.suppress(ValueError):
+                    self._websocket._dispatch.extra_events[name].remove(fut)
+                raise
+
+            if not check:
+                break
+            checked = check(*res)
+            if isawaitable(checked):
+                checked = await checked
+            if checked:
+                break
+            else:
+                # The check failed, so try again next time
+                log.info(f"A check failed waiting for the {name} event")
+
+        if res:
+            return res[0] if len(res) == 1 else res
+
+    async def wait_for_component(
+        self,
+        components: Union[
+            Union[str, Button, SelectMenu],
+            List[Union[str, Button, SelectMenu]],
+        ] = None,
+        messages: Union[Message, int, List[Union[Message, int]]] = None,
+        check: Optional[Callable[..., Union[bool, Awaitable[bool]]]] = None,
+        timeout: Optional[float] = None,
+    ):
+        """
+        Waits for a component to be interacted with, and returns the resulting context.
+
```

Comment:
```suggestion

.. information::
	If you are waiting for a select menu, you can find the selected values in ``ctx.data.values``

```
I'm unsure whether the thing is actually called `information` or something else, please look it up.

Also, does this work flawlessly if you use one of the new select types or does ctx.data.values end up being a list of dicts or something?

Reply:
> Also, does this work flawlessly if you use one of the new select types or does ctx.data.values end up being a list of dicts or something?
For the user select, it's a list of ids as strings. Adding it would be out of the scope of the pr, though it should be done before the next release